src/libs/ReportUtils.ts

```diff
@@ -4137,45 +4033,25 @@ function getIOUReportActionDisplayMessage(reportAction) {
  * - Not a public / admin / announce chat room (chat type doesn't match any of the specified types).
  * - More than 2 participants.
  *
- * @param {Object} report
- * @returns {Boolean}
  */
-function isGroupChat(report) {
-    return (
+function isGroupChat(report: OnyxEntry<Report>): boolean {
+    return Boolean(
         report &&
-        !isChatThread(report) &&
-        !isTaskReport(report) &&
-        !isMoneyRequestReport(report) &&
-        !isArchivedRoom(report) &&
-        !Object.values(CONST.REPORT.CHAT_TYPE).includes(getChatType(report)) &&
-        lodashGet(report, 'participantAccountIDs.length', 0) > 2
+            !isChatThread(report) &&
+            !isTaskReport(report) &&
+            !isMoneyRequestReport(report) &&
+            !isArchivedRoom(report) &&
+            !Object.values(CONST.REPORT.CHAT_TYPE).some((chatType) => chatType === getChatType(report)) &&
+            (report?.participantAccountIDs?.length ?? 0) > 2,
     );
 }
 
-/**
- * @param {Object} report
- * @returns {Boolean}
- */
-function isReportDraft(report) {
-    return isExpenseReport(report) && lodashGet(report, 'stateNum') === CONST.REPORT.STATE_NUM.OPEN && lodashGet(report, 'statusNum') === CONST.REPORT.STATUS.OPEN;
-}
-
-/**
- * @param {Object} report
- * @returns {Boolean}
- */
-function shouldUseFullTitleToDisplay(report) {
+function shouldUseFullTitleToDisplay(report: OnyxEntry<Report>): boolean {
     return isMoneyRequestReport(report) || isPolicyExpenseChat(report) || isChatRoom(report) || isChatThread(report) || isTaskReport(report);
 }
 
-/**
- *
- * @param {String} type
- * @param {String} policyID
- * @returns {Object}
- */
-function getRoom(type, policyID) {
-    const room = _.find(allReports, (report) => report && report.policyID === policyID && report.chatType === type && !isThread(report));
+function getRoom(type: ValueOf<typeof CONST.REPORT.CHAT_TYPE>, policyID: string) {
+    const room = Object.values(allReports ?? {}).find((report) => report && report.policyID === policyID && report.chatType === type && !isThread(report));
```

Comment:
```suggestion
    const room = Object.values(allReports ?? {}).find((report) => isNotEmptyObject(report) && report.policyID === policyID && report.chatType === type && !isThread(report));
```

Reply:
I dont think its needed here , I can remove check for report and do 
```suggestion
    const room = Object.values(allReports ?? {}).find((report) => report?.policyID === policyID && report?.chatType === type && !isThread(report));
```

Reply:
This is better